commands/version/index.js

```diff
@@ -232,7 +243,10 @@ class VersionCommand extends Command {
 
     // amending a commit probably means the working tree is dirty
     if (this.commitAndTag && this.gitOpts.amend !== true) {
-      tasks.unshift(() => checkWorkingTree(this.execOpts));
+      const { forcePublish, conventionalCommits, conventionalGraduate } = this.options;
+      const checkUncommittedOnly = forcePublish || (conventionalCommits && conventionalGraduate);
+      const check = checkUncommittedOnly ? checkWorkingTree.throwIfUncommitted : checkWorkingTree;
+      tasks.unshift(() => check(this.execOpts));
```

Comment:
These four lines are the fix for #1675.

Reply:
hrm, ok, this is confusing when conflated with the rest of the logic. I'd prefer this was in a separate PR for clarity.

Reply:
This PR isn't complete without these four lines - graduate prereleases is facilitated by passing `--conventional-graduate`, and that fails without this fix. It just so happens that this same issue is also a known bug for `--force-publish`.